payments_service/braintree/forms.py

```diff
@@ -23,3 +31,41 @@ def clean(self):
         if method_missing or too_many_methods:
             raise forms.ValidationError(
                 'Either pay_method_nonce or pay_method_uri can be submitted')
+
+
+class ChangeSubscriptionPayMethodForm(forms.Form):
+    new_pay_method_uri = forms.CharField(max_length=255)
+    subscription_uri = forms.CharField(max_length=255)
+
+    def __init__(self, user, *args, **kw):
+        self.user = user
+        super(ChangeSubscriptionPayMethodForm, self).__init__(*args, **kw)
+
+    def clean_subscription_uri(self):
+        uri = self.cleaned_data['subscription_uri']
+        if not self.user_owns_resource(uri,
+                                       'paymethod__braintree_buyer__buyer'):
+            raise forms.ValidationError(
+                'subscription by URI does not exist or belongs to another user'
+            )
```

Comment:
I guess so but then I'd have to do custom output formatting to render the error. A 400 bad request seems logical enough to me.